src/toast-container.tsx

```diff
@@ -1,217 +1,310 @@
+    /**
+     * Toggles toast container visibility
+     * @params show: boolean
+     * @returns 
+     */
+    toggle = (show: boolean) => {
+        this.setState({ visible: show });
+    } 
+    /**
+     * Shows a new toast
+     * @param message: string | JSX.Element
+     * @param toastOptions?: ToastOptions
+     * @returns id: string
+     */
+    show = (message: string | JSX.Element, toastOptions?: ToastOptions) => {
+        let id = toastOptions?.id || Math.random().toString();
+        const onDestroy = () => {
+            toastOptions?.onClose && toastOptions?.onClose();
+            if(this.state.foldedToast.id === id) return this.setState({ foldedToast: this.dummyToast})
+            this.setState({ toastsHistory: this.state.toastsHistory.filter((t) => t.id !== id) });
+        };
+
+        const newToast: ToastProps = {
             id,
             onDestroy,
             message,
             open: true,
             onHide: () => this.hide(id),
             ...this.props,
             ...toastOptions,
-          },
-          ...this.state.toasts.filter((t) => t.open),
-        ],
-      });
-    });
-
-    return id;
-  };
-
-  /**
-   * Updates a toast, To use this create you must pass an id to show method first, then pass it here to update the toast.
-   */
-  update = (
-    id: string,
-    message: string | JSX.Element,
-    toastOptions?: ToastOptions
-  ) => {
-    this.setState({
-      toasts: this.state.toasts.map((toast) =>
-        toast.id === id ? { ...toast, message, ...toastOptions } : toast
-      ),
-    });
-  };
-
-  /**
-   * Removes a toast from stack
-   */
-  hide = (id: string) => {
-    this.setState({
-      toasts: this.state.toasts.map((t) =>
-        t.id === id ? { ...t, open: false } : t
-      ),
-    });
-  };
-
-  /**
-   * Removes all toasts in stack
-   */
-  hideAll = () => {
-    this.setState({
-      toasts: this.state.toasts.map((t) => ({ ...t, open: false })),
-    });
-  };
-
-  /**
-   * Check if a toast is currently open
-   */
-  isOpen = (id: string) => {
-    return this.state.toasts.some((t) => t.id === id && t.open);
-  }
-
-  renderBottomToasts() {
-    const { toasts } = this.state;
-    let { offset, offsetBottom } = this.props;
-    let style: ViewStyle = {
-      bottom: offsetBottom || offset,
-      width: width,
-      justifyContent: "flex-end",
-      flexDirection: "column",
+        }
+
+        requestAnimationFrame(() => {
+            if(this.state.foldedToast === this.dummyToast) this.setState({ foldedToast: newToast }); 
+            else {
+                this.setState(
+                    {
+                        toastsHistory: [
+                            this.state.foldedToast,
+                            ...this.state.toastsHistory
+                        ],
+                        foldedToast: newToast
+                    }
+                )
+            }
+            });
+            return id;
+    };
+
+    renderUnfolded = () => {
```

Comment:
Name of the function is misleading, i would expect that this method is returning a JSX Element